tests/test_ptc.py

```diff
@@ -189,21 +208,29 @@ def ptcFitAndCheckPtc(self, order=None, fitType='', doTableArray=False):
                 self.assertAlmostEqual(np.sqrt(self.noiseSq), returnedDataset.noise[ampName])
 
             # Nonlinearity fit parameters
-            self.assertAlmostEqual(0.0, localDataset.nonLinearity[ampName][0])
-            self.assertAlmostEqual(self.flux, localDataset.nonLinearity[ampName][1])
-            self.assertAlmostEqual(self.k2NonLinearity, localDataset.nonLinearity[ampName][2])
+            self.assertAlmostEqual(0.0, returnedDataset.nonLinearity[ampName][0])
+            self.assertAlmostEqual(self.flux, returnedDataset.nonLinearity[ampName][1])
+            self.assertAlmostEqual(self.k2NonLinearity, returnedDataset.nonLinearity[ampName][2])
 
             # Non-linearity coefficient for linearizer
             self.assertAlmostEqual(-self.k2NonLinearity/(self.flux**2),
-                                   localDataset.coefficientLinearizeSquared[ampName])
+                                   returnedDataset.coefficientLinearizeSquared[ampName])
 
             # Linearity residuals
-            self.assertEqual(len(localDataset.nonLinearityResiduals[ampName]),
-                             len(self.inputNonLinearityResiduals[localDataset.visitMask[ampName]]))
+            self.assertEqual(len(returnedDataset.nonLinearityResiduals[ampName]),
+                             len(inputNonLinearityResiduals))
 
-            for i in np.arange(len(localDataset.nonLinearityResiduals[ampName])):
-                self.assertAlmostEqual(localDataset.nonLinearityResiduals[ampName][i],
-                                       self.inputNonLinearityResiduals[i])
+            for i in np.arange(len(returnedDataset.nonLinearityResiduals[ampName])):
```

Comment:
I wonder if there's a more elegant way of doing that?

Reply:
Maybe something like this?
`for calc, truth in zip(returnedDataset.nonLinearityResiduals[ampName], inputNonLinearityResiduals):
    self.assertAlmostEqual(calc, truth)`